hack/linter-extra.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -e -o pipefail
+
+if [ "$DISABLE_LINTER" == "true" ]
+then
+  exit 0
+fi
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if ! [ -x "$(command -v golangci-lint)" ]; then
+	echo "Installing GolangCI-Lint"
+	${DIR}/install_golint.sh -b $GOPATH/bin v1.20.0
+fi
+
+export GO111MODULE=on
```

Comment:
Personally, I find it hard to see what the difference between lint and lint-extra is. I fear this might confuse others as well. What do we gain? 

AFAICT, the bulk of the scripts are identical. So if we really want to split into multiple lint steps in `Makefile` and `jenkins-x-lint.yaml`, then I think it should be a single linter script and you pass in by argument which checks to run. 

If this is just about splitting into more tasks and contexts, one thing I want to do for a long time it to break out a `codegen` context (`jenkins-x-codegen.yaml) and so the code generation checks separately. That's technically really not linting and it has confused people before. I'd rather do that than splitting lint and lint-extra (unless there is something I am missing here)

Reply:
lint -> all the old linters, lint-extra -> new linters (ununsed and errcheck). It's a bit difficult for me to know where the script really fails. I tried various combinations, and this is the one which succeeded. I think it might make sense to coordinate over slack, if that's an option. I can try making the changes, and if you can get me the logs, that would help us debug it 🥇 wdyt? Let me try to separate the code gen from linting, and see what happens.